CMakeLists.txt

```diff
@@ -91,6 +91,8 @@ endif()
 if(NOT DNNL_TARGET_ARCH)
     if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*|arm64.*|ARM64.*)")
         set(DNNL_TARGET_ARCH "AARCH64")
+    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(ppc64.*|POWER.*)")
+        set(DNNL_TARGET_ARCH "POWER9")
```

Comment:
(probably, silly question) Why the target arch is `POWER9` and not generic `POWER` / `PPC64`?

Reply:
The question makes sense: I wrote POWER9 because that’s what I can test.
I’m perfectly happy changing it to either POWER or PPC64.

Reply:
I think it should be safe to assume that the code would work on any POWER architecture, since this is just a generic C++11 code. So, I would suggest to use version agnostic-name. If someone wants to contribute POWER9-specific optimizations later, we can always split and extend the knobs here.

I am not aware what is more adopted by the community for such kind of macros: `POWER` or `PPC64`. Quick googling showed that probably PPC64 probably suites better (given we use X64 and AARCH64 for x86_64 and Arm processors), and it is used in many major Linux distributions in their installation media names.
LLVM seems to use both versions: `__POWERPC64__`, `__POWERPC__`, `__PPC64__`. Not sure what the rules to choose are.

The only thing that bothers me is that it seems people sometimes distinguish between `ppc64` and `ppc64le`, meaning that `ppc64` is Big-Endian. I think we can say that `PPC64` is macro defined for all POWER architectures regardless of the endianness, and the code may additionally check for `__BYTE_ORDER__` or something.

Please let me know what you think.

Reply:
Yes, `PPC64` is more predictable: it matches the compiler macro defined by IBM’s XL C/C++, `_ARCH_PPC64`.
I prefer to spell it because it’s easier to scan the code looking for it, but it fits better with the others in its abbreviated form `PPC64` so I’m changing it. It’s also easier to find all related lines with grep.

The point of requiring `ppc64le` is that POWER8 is the first one to support that mode so you get a bunch of other assumptions with it which might be useful for other software. I have used my machine in BE mode a few times so I’d prefer not to restrict that unnecessarily: there are other constants like `__POWER8_VECTOR__` or `_ARCH_PWR8` we could use if needed to set POWER8 as a baseline.

There is a list of symbols used by different compilers here: https://sourceforge.net/p/predef/wiki/Architectures/#powerpc
At least for GCC, the specific macro for POWER9 is `_ARCH_PWR9` but it’s not defined in my system with gcc 9.3.0 although `_ARCH_PWR8` is there.

I’m changing the code to use `PPC64` consistently, then I’ll ammend the commit and force push to keep everything in one commit.

Reply:
Sounds good!
Thank you, @AlbertoGP!